hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallEvmCodesTest.java

```diff
@@ -138,9 +146,45 @@ void getBlockPrevrandao() {
         assertTrue(result.length() > "0x".length());
     }
 
+    @Test
+    void getBlockHashReturnsCorrectHash() {
+        // Persist all entities so that we can get a specific record file hash and pass it to
+        // functionEncodeDecoder#functionHashFor.
+        persistEntities();
+        areEntitiesPersisted = true;
+
+        final var functionHash = functionEncodeDecoder.functionHashFor(
+                "getBlockHash", EVM_CODES_ABI_PATH, recordFileForBlockHash.getIndex());
+        final var serviceParameters = serviceParametersForEvmCodes(functionHash);
+
+        assertThat(contractCallService.processCall(serviceParameters))
+                .isEqualTo(BlockHashUtil.ethHashFrom(recordFileForBlockHash.getHash())
+                        .toString());
```

Comment:
No need for the util.
```suggestion
                .isEqualTo(recordFileForBlockHash.getHash().substring(0, 64));
```

Reply:
Comment addressed.